migrations/src/main/scala/za/co/absa/enceladus/migrations/framework/Migrator.scala

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.migrations.framework
+
+import org.apache.log4j.{LogManager, Logger}
+
+import util.control.Breaks._
+import za.co.absa.enceladus.migrations.framework.dao.DocumentDb
+import za.co.absa.enceladus.migrations.framework.migration._
+
+class Migrator(db: DocumentDb, migrations: Seq[Migration]) {
+
+  private val log: Logger = LogManager.getLogger("CollectionMigration")
+
+  def getCollectionMigrations: Seq[CollectionMigration] = migrations.collect({ case m: CollectionMigration => m })
+
+  def getQueryMigrations: Seq[CommandMigration] = migrations.collect({ case m: CommandMigration => m })
+
+  def getJsonMigrations: Seq[JsonMigration] = migrations.collect({ case m: JsonMigration => m })
+
+  /**
+    * Do the migration from the current version of the database to the target one.
+    *
+    * The migrations passed into the constructor should be:
+    * - In the order of database versions
+    * - Without gaps in version numbers
+    * - There should be only one migration per version switch
+    *
+    * @param targetDbVersion A version of the database to migrate to.
+    */
+  def migrate(targetDbVersion: Int): Unit = {
+    validate(targetDbVersion)
+
+    val sourceDbVersion = db.getVersion()
+    validateDbVersios(sourceDbVersion, targetDbVersion)
+
+    var currentVersionCollections = getCollectionNames(sourceDbVersion)
+
+    if (currentVersionCollections.isEmpty) {
+      throw new IllegalStateException(s"No collection names are registered for db version $sourceDbVersion.")
+    }
+
+    if (targetDbVersion > sourceDbVersion) {
```

Comment:
I think it would not even get to this point if this would not be true. There are a lot of validations beforehand. Like line 46 of `MigrationService.scala`

Reply:
If `targetDbVersion == sourceDbVersion` all validations should pass and no migrations should run. I'll take a look if the `for(...)` below will work just fine without the `if`.